druid/examples/cursor.rs

```diff
@@ -61,11 +45,17 @@ impl<W: Widget<AppState>> Controller<AppState, W> for CursorArea {
         data: &mut AppState,
         env: &Env,
     ) {
-        if data.custom.is_none() {
-            data.custom = ctx.window().make_cursor(&data.custom_desc).map(Arc::new);
-        }
-        if matches!(event, Event::MouseMove(_)) {
-            ctx.set_cursor(&data.cursor);
+        match event {
+            Event::WindowConnected => {
+                let cursor_image =
+                    ImageBuf::from_data(include_bytes!("./assets/PicWithAlpha.png")).unwrap();
```

Comment:
while I agree that it makes sense to move the creation of the actual cursor into `WindowConnected`, I would leave the logic around creating the `ImageBuf` where it was.

How I would think about this is in terms of a principle that may have a name, but is basically: "all else being equal, state should be declared at the earliest opportunity." If something is known at compile time, and can be `const`, it should be `const`; if something can be computed in `main` it should be in `main`; etc. There are of course exceptions, such as if something has runtime or binary-size costs, but I think this is a good rule of thumb because it makes code easier to reason about; you can assume that if something is being done in a given place it is because it can't be done earlier, which makes it easier to reason about how a program works.

Reply:
I get the point, I thought that since the variable would only be used once, it seems a bit useless to use the `data` for that, ad usually the state goes into the `data`. But I also see you point of doing as much as early as possible.

Reply:
I agree the data might not be perfect, and I would probably put a `#[data(ignore)]` annotation on that field? I think in a perfect world it would actually be *widget* state...

Reply:
It cannot be in a widgets state because both the button and the `CursorArea` need acces to it, I can create a custom widget/controller but that would be very elaborate. IMO the perfect would be one where `CursorDesc::new` and `ImageBuf::from_data` are const, but they are not.

also should I open an issue about documenting `#[data(ignore)]`? because it seems very useful for things like this, fields that are static/never change but need to be known to multiple widgets.

Reply:
Just in general, should I open documentation issues? Maybe its worth having one issue with a checklist instead of a bunch of small issues.

Reply:
Opening an issue for that sounds good. In my experience, multiple small issues work better than large checklists, and feel more satisfying to close :)

Reply:
I can relate to that